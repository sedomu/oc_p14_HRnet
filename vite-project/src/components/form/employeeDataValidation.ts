export function employeeDataValidation(data: any) {
    const errors: Record<string, string> = {};


    console.log("j'ai bien un data");
    console.log(data);

    if(!data.firstName || data.firstName.length < 2){
        errors.firstName = "First name must be at least 2 characters long";
    }

    console.log(errors);

    return errors;
}