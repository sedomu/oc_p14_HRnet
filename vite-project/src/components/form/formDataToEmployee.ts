import type { Employee } from "@/redux.ts";

export default function formDataToEmployee(formData: FormData): Employee {
    const firstName = formData.get("firstName")?.toString();
    const lastName = formData.get("lastName")?.toString();
    const dateOfBirth = formData.get("dateOfBirth")?.toString();
    const startDate = formData.get("startDate")?.toString();
    const street = formData.get("street")?.toString();
    const city = formData.get("city")?.toString();
    const state = formData.get("state")?.toString();
    const zipCode = Number(formData.get("zipCode"));
    const department = formData.get("department")?.toString();

    return {
        firstName: firstName ? firstName : "",
        lastName: lastName ? lastName : "",
        dateOfBirth: dateOfBirth ? dateOfBirth : "",
        startDate: startDate ? startDate : "",
        street: street ? street : "",
        city: city ? city : "",
        state: state ? state : "",
        zipCode: zipCode ? zipCode : 0,
        department: department ? department : "",
    };
}
