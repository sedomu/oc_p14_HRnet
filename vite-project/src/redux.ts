import {configureStore, createSlice, type PayloadAction} from '@reduxjs/toolkit'
import {MOCKUP_DATA} from "@/config.ts";
import {employeesMockupData} from "@/data/employees_mockup.ts";

export type Employee = {
    firstName: string
    lastName: string
    dateOfBirth: string
    startDate: string
    street: string
    city: string
    state: string
    zipCode: number
    department: string
}

const initialState: Employee[] = []

if (MOCKUP_DATA){
    initialState.push(...employeesMockupData);
}

const employeesSlice = createSlice({
    name: 'employees',
    initialState: initialState,
    reducers: {
        addEmployee: (state, action: PayloadAction<Employee>) => {
            state.push(action.payload)
        }
    }
})

export const store = configureStore({
    reducer: {
        employees: employeesSlice.reducer
    }
})

export const { addEmployee } = employeesSlice.actions

export type RootState = ReturnType<typeof store.getState>;