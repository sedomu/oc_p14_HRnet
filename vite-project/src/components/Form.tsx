import {cn} from "@/lib/utils"
import {Button} from "@/components/ui/button"
import {Card, CardContent, CardHeader, CardTitle,} from "@/components/ui/card"
import {Input} from "@/components/ui/input"
import {Label} from "@/components/ui/label"
import {DatePicker} from "@/components/DatePicker.tsx";
import {ListSelect} from "@/components/ListSelect.tsx";
import type {FormEvent} from "react";
import {useDispatch} from "react-redux";
import {addEmployee, type Employee} from "@/redux.ts";

type formProps = {
    className?: string; onSuccess: () => void;
}

function formDataToEmployee(formData: FormData): Employee | null {
    const firstName = formData.get("firstName")?.toString();
    const lastName = formData.get("lastName")?.toString();
    const dateOfBirth = formData.get("dateOfBirth")?.toString();
    const startDate = formData.get("startDate")?.toString();
    const street = formData.get("street")?.toString();
    const city = formData.get("city")?.toString();
    const state = formData.get("state")?.toString();
    const zipCodeStr = formData.get("zipCode")?.toString();
    const department = formData.get("department")?.toString();

    // VÃ©rification des champs obligatoires
    if (!firstName || !lastName || !dateOfBirth || !startDate || !street || !city || !state || !zipCodeStr || !department) {
        return null;
    }

    const zipCode = Number(zipCodeStr);
    if (Number.isNaN(zipCode)) return null;

    return {
        firstName,
        lastName,
        dateOfBirth,
        startDate,
        street,
        city,
        state,
        zipCode,
        department,
    };
}

export function Form({className, onSuccess}: formProps) {

    const states = [{
        "name": "Alabama", "key": "AL"
    }, {
        "name": "Alaska", "key": "AK"
    }, {
        "name": "American Samoa", "key": "AS"
    }, {
        "name": "Arizona", "key": "AZ"
    }, {
        "name": "Arkansas", "key": "AR"
    }, {
        "name": "California", "key": "CA"
    }, {
        "name": "Colorado", "key": "CO"
    }, {
        "name": "Connecticut", "key": "CT"
    }, {
        "name": "Delaware", "key": "DE"
    }, {
        "name": "District Of Columbia", "key": "DC"
    }, {
        "name": "Federated States Of Micronesia", "key": "FM"
    }, {
        "name": "Florida", "key": "FL"
    }, {
        "name": "Georgia", "key": "GA"
    }, {
        "name": "Guam", "key": "GU"
    }, {
        "name": "Hawaii", "key": "HI"
    }, {
        "name": "Idaho", "key": "ID"
    }, {
        "name": "Illinois", "key": "IL"
    }, {
        "name": "Indiana", "key": "IN"
    }, {
        "name": "Iowa", "key": "IA"
    }, {
        "name": "Kansas", "key": "KS"
    }, {
        "name": "Kentucky", "key": "KY"
    }, {
        "name": "Louisiana", "key": "LA"
    }, {
        "name": "Maine", "key": "ME"
    }, {
        "name": "Marshall Islands", "key": "MH"
    }, {
        "name": "Maryland", "key": "MD"
    }, {
        "name": "Massachusetts", "key": "MA"
    }, {
        "name": "Michigan", "key": "MI"
    }, {
        "name": "Minnesota", "key": "MN"
    }, {
        "name": "Mississippi", "key": "MS"
    }, {
        "name": "Missouri", "key": "MO"
    }, {
        "name": "Montana", "key": "MT"
    }, {
        "name": "Nebraska", "key": "NE"
    }, {
        "name": "Nevada", "key": "NV"
    }, {
        "name": "New Hampshire", "key": "NH"
    }, {
        "name": "New Jersey", "key": "NJ"
    }, {
        "name": "New Mexico", "key": "NM"
    }, {
        "name": "New York", "key": "NY"
    }, {
        "name": "North Carolina", "key": "NC"
    }, {
        "name": "North Dakota", "key": "ND"
    }, {
        "name": "Northern Mariana Islands", "key": "MP"
    }, {
        "name": "Ohio", "key": "OH"
    }, {
        "name": "Oklahoma", "key": "OK"
    }, {
        "name": "Oregon", "key": "OR"
    }, {
        "name": "Palau", "key": "PW"
    }, {
        "name": "Pennsylvania", "key": "PA"
    }, {
        "name": "Puerto Rico", "key": "PR"
    }, {
        "name": "Rhode Island", "key": "RI"
    }, {
        "name": "South Carolina", "key": "SC"
    }, {
        "name": "South Dakota", "key": "SD"
    }, {
        "name": "Tennessee", "key": "TN"
    }, {
        "name": "Texas", "key": "TX"
    }, {
        "name": "Utah", "key": "UT"
    }, {
        "name": "Vermont", "key": "VT"
    }, {
        "name": "Virgin Islands", "key": "VI"
    }, {
        "name": "Virginia", "key": "VA"
    }, {
        "name": "Washington", "key": "WA"
    }, {
        "name": "West Virginia", "key": "WV"
    }, {
        "name": "Wisconsin", "key": "WI"
    }, {
        "name": "Wyoming", "key": "WY"
    }];
    const departments = [{
        "name": "Sales", "key": "Sales"
    }, {
        "name": "Marketing", "key": "Marketing"
    }, {
        "name": "Engineering", "key": "Engineering"
    }, {
        "name": "Human Resources", "key": "Human Resources"
    }, {
        "name": "Legal", "key": "Legal"
    },];

    const dispatch = useDispatch();

    const handleSubmit = (e: FormEvent<HTMLFormElement>) => {
        e.preventDefault();

        const formData = new FormData(e.currentTarget);

        const data = formDataToEmployee(formData);

        if (!data) {
            console.error("Formulaire invalide");
            return;
        }

        dispatch(addEmployee(data));

        onSuccess();


    }

    return (<div className={cn("flex flex-col gap-6", className)}>
        <Card>
            <CardHeader className="text-center">
                <CardTitle className="text-xl">Create Employee</CardTitle>
            </CardHeader>
            <CardContent>
                <form onSubmit={handleSubmit}>
                    <div className="grid gap-6">

                        <div className="grid gap-6">
                            <div className="grid gap-3">
                                <Label htmlFor="firstName">First
                                    Name</Label>
                                <Input
                                    id="firstName"
                                    type="text"
                                    placeholder=""
                                    name="firstName"
                                    required
                                />
                            </div>
                            <div className="grid gap-3">
                                <Label htmlFor="lastName">Last Name</Label>
                                <Input
                                    id="lastName"
                                    type="text"
                                    placeholder=""
                                    name="lastName"
                                    required
                                />
                            </div>
                            <div className="grid gap-3">
                                <Label htmlFor="dateOfBirth">Date of
                                    Birth</Label>
                                <DatePicker name="dateOfBirth"/>
                            </div>
                            <div className="grid gap-3">
                                <Label htmlFor="startDate">Start
                                    Date</Label>
                                <DatePicker name="startDate"/>
                            </div>
                            <div
                                className="after:border-border relative text-center text-sm after:absolute after:inset-0 after:top-1/2 after:z-0 after:flex after:items-center after:border-t">
                                    <span
                                        className="bg-card text-muted-foreground relative z-10 px-2">
                                        Address
                                    </span>
                            </div>

                            <div className="grid gap-3">
                                <Label htmlFor="street">Street</Label>
                                <Input
                                    id="street"
                                    type="text"
                                    placeholder=""
                                    name="street"
                                    required
                                />
                            </div>

                            <div className="grid gap-3">
                                <Label htmlFor="city">City</Label>
                                <Input
                                    id="city"
                                    type="text"
                                    placeholder=""
                                    name="city"
                                    required
                                />
                            </div>

                            <div className="grid gap-3">
                                <Label htmlFor="state">State</Label>
                                <ListSelect name="state" options={states}/>
                            </div>

                            <div className="grid gap-3">
                                <Label htmlFor="zipCode">Zip Code</Label>
                                <Input
                                    id="zipCode"
                                    type="number"
                                    name="zipCode"
                                    placeholder=""
                                    min={0}
                                    required
                                    className=" file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive appearance-none"
                                    style={{
                                        WebkitAppearance: "none", // Chrome, Safari, Edge
                                        MozAppearance: "textfield", // Firefox
                                    }}
                                />
                            </div>

                            <div
                                className="after:border-border relative text-center text-sm after:absolute after:inset-0 after:top-1/2 after:z-0 after:flex after:items-center after:border-t"></div>

                            <div className="grid gap-3">
                                <Label
                                    htmlFor="department">Department</Label>
                                <ListSelect name="department"
                                            options={departments}/>
                            </div>


                            <Button type="submit" className="w-full">
                                Save
                            </Button>
                        </div>

                    </div>
                </form>
            </CardContent>
        </Card>

    </div>);
}
