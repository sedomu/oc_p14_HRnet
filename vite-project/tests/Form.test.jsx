import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import { Form } from "@/components/form/Form";
import { Provider } from "react-redux";
import { store } from "@/redux";

import { userEvent } from "@testing-library/user-event";

describe("<Form/>", () => {
    it("should render title", () => {
        render(
            <Provider store={store}>
                <Form onSuccess={() => {}}></Form>
            </Provider>
        );
        const createEmployeeTitle = screen.getByText("Create Employee");
        expect(createEmployeeTitle.textContent).toBe("Create Employee");
    });
    it("should update redux's store", async () => {
        render(
            <Provider store={store}>
                <Form onSuccess={() => {}}></Form>
            </Provider>
        );

        await userEvent.type(screen.getByLabelText("First Name"), "John");
        await userEvent.type(screen.getByLabelText("Last Name"), "Doe");
        await userEvent.type(screen.getByTestId("dateOfBirth"), "2000-01-01");
        await userEvent.type(screen.getByTestId("startDate"), "2022-01-01");
        await userEvent.type(
            screen.getByLabelText("Street"),
            "123 Main street"
        );
        await userEvent.type(screen.getByLabelText("City"), "Anytown");
        await userEvent.click(screen.getByText("New York"));
        await userEvent.type(screen.getByLabelText("Zip Code"), "12345");
        await userEvent.click(screen.getByText("Sales"));

        await userEvent.click(screen.getByText("Save"));

        const state = store.getState();
        expect(state.employees.length).toBe(1);
    });
});
