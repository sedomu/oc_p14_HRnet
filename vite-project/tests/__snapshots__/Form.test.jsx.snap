// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Form/> > should render correctly 1`] = `
<div
  class="flex flex-col gap-6"
>
  <div
    class="bg-card text-card-foreground flex flex-col gap-6 rounded-xl border py-6 shadow-sm"
    data-slot="card"
  >
    <div
      class="@container/card-header grid auto-rows-min grid-rows-[auto_auto] items-start gap-1.5 px-6 has-data-[slot=card-action]:grid-cols-[1fr_auto] [.border-b]:pb-6 text-center"
      data-slot="card-header"
    >
      <div
        class="font-semibold text-xl"
        data-slot="card-title"
      >
        Create Employee
      </div>
    </div>
    <div
      class="px-6"
      data-slot="card-content"
    >
      <form>
        <div
          class="grid gap-6"
        >
          <div
            class="grid gap-6"
          >
            <div
              class="grid gap-3"
            >
              <label
                class="flex items-center gap-2 text-sm leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50"
                data-slot="label"
                for="firstName"
              >
                First Name
              </label>
              <input
                class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive"
                data-slot="input"
                id="firstName"
                name="firstName"
                placeholder=""
                required=""
                type="text"
              />
            </div>
            <div
              class="grid gap-3"
            >
              <label
                class="flex items-center gap-2 text-sm leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50"
                data-slot="label"
                for="lastName"
              >
                Last Name
              </label>
              <input
                class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive"
                data-slot="input"
                id="lastName"
                name="lastName"
                placeholder=""
                required=""
                type="text"
              />
            </div>
            <div
              class="grid gap-3"
            >
              <label
                class="flex items-center gap-2 text-sm leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50"
                data-slot="label"
                for="dateOfBirth"
              >
                Date of Birth
              </label>
              <button
                aria-controls="radix-«r0»"
                aria-expanded="false"
                aria-haspopup="dialog"
                class="inline-flex items-center gap-2 whitespace-nowrap rounded-md text-sm transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive border bg-background shadow-xs hover:bg-accent hover:text-accent-foreground dark:bg-input/30 dark:border-input dark:hover:bg-input/50 h-9 px-4 py-2 has-[>svg]:px-3 w-full justify-between font-normal"
                data-slot="popover-trigger"
                data-state="closed"
                id="date"
                type="button"
              >
                <span />
                <svg
                  aria-hidden="true"
                  class="lucide lucide-chevron-down"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="m6 9 6 6 6-6"
                  />
                </svg>
              </button>
              <input
                name="dateOfBirth"
                type="hidden"
                value=""
              />
            </div>
            <div
              class="grid gap-3"
            >
              <label
                class="flex items-center gap-2 text-sm leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50"
                data-slot="label"
                for="startDate"
              >
                Start Date
              </label>
              <button
                aria-controls="radix-«r1»"
                aria-expanded="false"
                aria-haspopup="dialog"
                class="inline-flex items-center gap-2 whitespace-nowrap rounded-md text-sm transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive border bg-background shadow-xs hover:bg-accent hover:text-accent-foreground dark:bg-input/30 dark:border-input dark:hover:bg-input/50 h-9 px-4 py-2 has-[>svg]:px-3 w-full justify-between font-normal"
                data-slot="popover-trigger"
                data-state="closed"
                id="date"
                type="button"
              >
                <span />
                <svg
                  aria-hidden="true"
                  class="lucide lucide-chevron-down"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="m6 9 6 6 6-6"
                  />
                </svg>
              </button>
              <input
                name="startDate"
                type="hidden"
                value=""
              />
            </div>
            <div
              class="after:border-border relative text-center text-sm after:absolute after:inset-0 after:top-1/2 after:z-0 after:flex after:items-center after:border-t"
            >
              <span
                class="bg-card text-muted-foreground relative z-10 px-2"
              >
                Address
              </span>
            </div>
            <div
              class="grid gap-3"
            >
              <label
                class="flex items-center gap-2 text-sm leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50"
                data-slot="label"
                for="street"
              >
                Street
              </label>
              <input
                class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive"
                data-slot="input"
                id="street"
                name="street"
                placeholder=""
                required=""
                type="text"
              />
            </div>
            <div
              class="grid gap-3"
            >
              <label
                class="flex items-center gap-2 text-sm leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50"
                data-slot="label"
                for="city"
              >
                City
              </label>
              <input
                class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive"
                data-slot="input"
                id="city"
                name="city"
                placeholder=""
                required=""
                type="text"
              />
            </div>
            <div
              class="grid gap-3"
            >
              <label
                class="flex items-center gap-2 text-sm leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50"
                data-slot="label"
                for="state"
              >
                State
              </label>
              <button
                aria-autocomplete="none"
                aria-controls="radix-«r2»"
                aria-expanded="false"
                class="border-input data-[placeholder]:text-muted-foreground [&_svg:not([class*='text-'])]:text-muted-foreground focus-visible:border-ring focus-visible:ring-ring/50 aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive dark:bg-input/30 dark:hover:bg-input/50 flex items-center justify-between gap-2 rounded-md border bg-transparent px-3 py-2 text-sm whitespace-nowrap shadow-xs transition-[color,box-shadow] outline-none focus-visible:ring-[3px] disabled:cursor-not-allowed disabled:opacity-50 data-[size=default]:h-9 data-[size=sm]:h-8 *:data-[slot=select-value]:line-clamp-1 *:data-[slot=select-value]:flex *:data-[slot=select-value]:items-center *:data-[slot=select-value]:gap-2 [&_svg]:pointer-events-none [&_svg]:shrink-0 [&_svg:not([class*='size-'])]:size-4 w-full"
                data-placeholder=""
                data-size="default"
                data-slot="select-trigger"
                data-state="closed"
                dir="ltr"
                role="combobox"
                type="button"
              >
                <span
                  data-slot="select-value"
                  style="pointer-events: none;"
                />
                <svg
                  aria-hidden="true"
                  class="lucide lucide-chevron-down size-4 opacity-50"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="m6 9 6 6 6-6"
                  />
                </svg>
              </button>
              <select
                aria-hidden="true"
                name="state"
                style="position: absolute; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;"
                tabindex="-1"
              >
                <option
                  value=""
                />
                <option
                  value="AL"
                >
                  Alabama
                </option>
                <option
                  value="AK"
                >
                  Alaska
                </option>
                <option
                  value="AS"
                >
                  American Samoa
                </option>
                <option
                  value="AZ"
                >
                  Arizona
                </option>
                <option
                  value="AR"
                >
                  Arkansas
                </option>
                <option
                  value="CA"
                >
                  California
                </option>
                <option
                  value="CO"
                >
                  Colorado
                </option>
                <option
                  value="CT"
                >
                  Connecticut
                </option>
                <option
                  value="DE"
                >
                  Delaware
                </option>
                <option
                  value="DC"
                >
                  District Of Columbia
                </option>
                <option
                  value="FM"
                >
                  Federated States Of Micronesia
                </option>
                <option
                  value="FL"
                >
                  Florida
                </option>
                <option
                  value="GA"
                >
                  Georgia
                </option>
                <option
                  value="GU"
                >
                  Guam
                </option>
                <option
                  value="HI"
                >
                  Hawaii
                </option>
                <option
                  value="ID"
                >
                  Idaho
                </option>
                <option
                  value="IL"
                >
                  Illinois
                </option>
                <option
                  value="IN"
                >
                  Indiana
                </option>
                <option
                  value="IA"
                >
                  Iowa
                </option>
                <option
                  value="KS"
                >
                  Kansas
                </option>
                <option
                  value="KY"
                >
                  Kentucky
                </option>
                <option
                  value="LA"
                >
                  Louisiana
                </option>
                <option
                  value="ME"
                >
                  Maine
                </option>
                <option
                  value="MH"
                >
                  Marshall Islands
                </option>
                <option
                  value="MD"
                >
                  Maryland
                </option>
                <option
                  value="MA"
                >
                  Massachusetts
                </option>
                <option
                  value="MI"
                >
                  Michigan
                </option>
                <option
                  value="MN"
                >
                  Minnesota
                </option>
                <option
                  value="MS"
                >
                  Mississippi
                </option>
                <option
                  value="MO"
                >
                  Missouri
                </option>
                <option
                  value="MT"
                >
                  Montana
                </option>
                <option
                  value="NE"
                >
                  Nebraska
                </option>
                <option
                  value="NV"
                >
                  Nevada
                </option>
                <option
                  value="NH"
                >
                  New Hampshire
                </option>
                <option
                  value="NJ"
                >
                  New Jersey
                </option>
                <option
                  value="NM"
                >
                  New Mexico
                </option>
                <option
                  value="NY"
                >
                  New York
                </option>
                <option
                  value="NC"
                >
                  North Carolina
                </option>
                <option
                  value="ND"
                >
                  North Dakota
                </option>
                <option
                  value="MP"
                >
                  Northern Mariana Islands
                </option>
                <option
                  value="OH"
                >
                  Ohio
                </option>
                <option
                  value="OK"
                >
                  Oklahoma
                </option>
                <option
                  value="OR"
                >
                  Oregon
                </option>
                <option
                  value="PW"
                >
                  Palau
                </option>
                <option
                  value="PA"
                >
                  Pennsylvania
                </option>
                <option
                  value="PR"
                >
                  Puerto Rico
                </option>
                <option
                  value="RI"
                >
                  Rhode Island
                </option>
                <option
                  value="SC"
                >
                  South Carolina
                </option>
                <option
                  value="SD"
                >
                  South Dakota
                </option>
                <option
                  value="TN"
                >
                  Tennessee
                </option>
                <option
                  value="TX"
                >
                  Texas
                </option>
                <option
                  value="UT"
                >
                  Utah
                </option>
                <option
                  value="VT"
                >
                  Vermont
                </option>
                <option
                  value="VI"
                >
                  Virgin Islands
                </option>
                <option
                  value="VA"
                >
                  Virginia
                </option>
                <option
                  value="WA"
                >
                  Washington
                </option>
                <option
                  value="WV"
                >
                  West Virginia
                </option>
                <option
                  value="WI"
                >
                  Wisconsin
                </option>
                <option
                  value="WY"
                >
                  Wyoming
                </option>
              </select>
            </div>
            <div
              class="grid gap-3"
            >
              <label
                class="flex items-center gap-2 text-sm leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50"
                data-slot="label"
                for="zipCode"
              >
                Zip Code
              </label>
              <input
                class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive appearance-none"
                data-slot="input"
                id="zipCode"
                min="0"
                name="zipCode"
                placeholder=""
                required=""
                type="number"
              />
            </div>
            <div
              class="after:border-border relative text-center text-sm after:absolute after:inset-0 after:top-1/2 after:z-0 after:flex after:items-center after:border-t"
            />
            <div
              class="grid gap-3"
            >
              <label
                class="flex items-center gap-2 text-sm leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50"
                data-slot="label"
                for="department"
              >
                Department
              </label>
              <button
                aria-autocomplete="none"
                aria-controls="radix-«r3»"
                aria-expanded="false"
                class="border-input data-[placeholder]:text-muted-foreground [&_svg:not([class*='text-'])]:text-muted-foreground focus-visible:border-ring focus-visible:ring-ring/50 aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive dark:bg-input/30 dark:hover:bg-input/50 flex items-center justify-between gap-2 rounded-md border bg-transparent px-3 py-2 text-sm whitespace-nowrap shadow-xs transition-[color,box-shadow] outline-none focus-visible:ring-[3px] disabled:cursor-not-allowed disabled:opacity-50 data-[size=default]:h-9 data-[size=sm]:h-8 *:data-[slot=select-value]:line-clamp-1 *:data-[slot=select-value]:flex *:data-[slot=select-value]:items-center *:data-[slot=select-value]:gap-2 [&_svg]:pointer-events-none [&_svg]:shrink-0 [&_svg:not([class*='size-'])]:size-4 w-full"
                data-placeholder=""
                data-size="default"
                data-slot="select-trigger"
                data-state="closed"
                dir="ltr"
                role="combobox"
                type="button"
              >
                <span
                  data-slot="select-value"
                  style="pointer-events: none;"
                />
                <svg
                  aria-hidden="true"
                  class="lucide lucide-chevron-down size-4 opacity-50"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="m6 9 6 6 6-6"
                  />
                </svg>
              </button>
              <select
                aria-hidden="true"
                name="department"
                style="position: absolute; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;"
                tabindex="-1"
              >
                <option
                  value=""
                />
                <option
                  value="Sales"
                >
                  Sales
                </option>
                <option
                  value="Marketing"
                >
                  Marketing
                </option>
                <option
                  value="Engineering"
                >
                  Engineering
                </option>
                <option
                  value="Human Resources"
                >
                  Human Resources
                </option>
                <option
                  value="Legal"
                >
                  Legal
                </option>
              </select>
            </div>
            <button
              class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive bg-primary text-primary-foreground shadow-xs hover:bg-primary/90 h-9 px-4 py-2 has-[>svg]:px-3 w-full"
              data-slot="button"
              type="submit"
            >
              Save
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
